blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DatastaxIO.java

```diff
@@ -0,0 +1,147 @@
+    private static final IOConfig ioconfig = IOConfig.singleton();
+
+    private static  Cluster cluster;
+    private static Session session;
+
+//    private final static Meter hostMeter = utils.Metrics.meter(DatastaxIO.class, "Hosts Connected");
+//    private final static Meter openConnectionsMeter = utils.Metrics.meter(DatastaxIO.class, "Total Open Connections");
+//    private final static Meter inFlightQueriesMeter = utils.Metrics.meter(DatastaxIO.class, "Total InFlight Queries");
+//    private final static Meter trashedConnectionsMeter = utils.Metrics.meter(DatastaxIO.class, "Total Trashed Connections");
+//    private final static Meter maxLoadMeter = utils.Metrics.meter(DatastaxIO.class, "Maximum Load");
+
+    static {
+        connect();
+        monitorConnection();
+    }
+
+    public static DatastaxIO singleton() { return INSTANCE; }
+
+    private DatastaxIO() {
+    }
+
+    private static void connect() {
+        Set<InetSocketAddress> dbHosts = ioconfig.getUniqueBinaryTransportHostsAsInetSocketAddresses();
+
+        CodecRegistry codecRegistry = new CodecRegistry();
+
+        cluster = Cluster.builder()
+                .withLoadBalancingPolicy(new TokenAwarePolicy(DCAwareRoundRobinPolicy.builder().withLocalDc("datacenter1").build()))
+                .withPoolingOptions(getPoolingOptions(dbHosts.size()))
+                .withCodecRegistry(codecRegistry)
+                .withSocketOptions(getSocketOptions())
+                .addContactPointsWithPorts(dbHosts)
+                .build();
+
+        if ( LOG.isDebugEnabled() ) {
+            logDebugConnectionInfo();
+        }
+
+        try {
+            session = cluster.connect();
+        }
+        catch (NoHostAvailableException e){
+            // TODO: figure out how to bubble this up
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void logDebugConnectionInfo() {
+        if ( cluster == null ) {
+            throw new IllegalStateException("cluster is not initialized");
+        }
+        final Metadata metadata = cluster.getMetadata();
+        LOG.debug("Connected to cluster: " + metadata.getClusterName());
+        for (final Host host : metadata.getAllHosts()) {
+            LOG.debug(String.format("Datacenter: %s; Host: %s; Rack: %s\n",
+                      host.getDatacenter(), host.getAddress(), host.getRack()));
+        }
+    }
+
+    private static SocketOptions getSocketOptions() {
+        final SocketOptions socketOptions = new SocketOptions();
+        socketOptions.setConnectTimeoutMillis(ioconfig.getRequestTimeout())
+                     .setReadTimeoutMillis(ioconfig.getRequestTimeout());
+        return socketOptions;
+    }
+
+    private static PoolingOptions getPoolingOptions(int numHosts){
+
+        final PoolingOptions poolingOptions = new PoolingOptions();
+        poolingOptions
+                .setCoreConnectionsPerHost(HostDistance.LOCAL, 15)
```

Comment:
I can, yes, let me do that. The Remote thing is not quite relevant to us yet. That's for when we are cross DC. I will try to remove the .setRemote() from code.